### PR TITLE
fix: useOpenAction supports sponsoredApproved, signlessApproved flags

### PR DESCRIPTION
**Pull request by cesarenaldi**

### CODE REVIEW

File: packages/react/src/transactions/useFollow.ts [hunk 1]
```diff
@@ -24,82 +24,69 @@ import {
...
+    session?.authenticated,
+    'You must be authenticated to use this operation. Use `useLogin` hook to authenticate.',
+  );
+  invariant(
+    session.type === SessionType.WithProfile,
+    'You must have a profile to use this operation.',
+  );
 
-      case FollowPolicyType.UNKNOWN:
-        const result = await fetchModule({ implementation: followPolicy.contractAddress });
```
Comment (krzysu) [inline packages/react/src/transactions/useFollow.ts:73]: no fetching module data for unknown module anymore?

File: packages/react/src/transactions/useFollow.ts [hunk 2]
```diff
@@ -24,82 +24,69 @@ import {
...
-        return {
-          kind: TransactionKind.FOLLOW_PROFILE,
-          profileId: args.profile.id,
-          signless: session.profile.signless,
-          sponsored: args.sponsored ?? true,
-        };
+/**
+ * @internal
+ */
+export function createFollowRequest(args: FollowArgs, session?: Session): FollowRequest {
```
Comment (krzysu) [inline packages/react/src/transactions/useFollow.ts:37]: does it need to be exported? I don't see it used anywhere outside of this file

File: packages/react/src/transactions/useBlockProfiles.ts [hunk 3]
```diff
@@ -34,9 +34,10 @@ export type BlockProfileArgs = {
    * - {@link BroadcastingErrorReason.RATE_LIMITED} - the profile reached the rate limit
    * - {@link BroadcastingErrorReason.APP_NOT_ALLOWED} - the app is not whitelisted for gasless transactions
    *
-   * @defaultValue true, the request will be attempted to be sponsored by the Lens API.
+   * If not specified, or `true`, the hook will attempt a Signless Experience when possible;
+   * otherwise, it will fall back to a signed experience.
    */
-  sponsored?: boolean;
+  sponsored?: false;
```
Comment (krzysu) [inline packages/react/src/transactions/useBlockProfiles.ts:40]: is `false` here intentional?

File: packages/react/src/transactions/useFollow.ts (hunk 1 above)
Comment (cesarenaldi) [reply packages/react/src/transactions/useFollow.ts:73]: It is not needed when following, the information I was looking for (i.e. `sponsoredApproved` and `signlessApproved`) is repeated in the `UnknownFollowModuleSettings` for convenience.

On the other hand it's going to be needed when configuring a Profile's Follow Module and when creating a new publication.

File: packages/react/src/transactions/useFollow.ts (hunk 2 above)
Comment (cesarenaldi) [reply packages/react/src/transactions/useFollow.ts:37]: no more, cleared some of this in the following PR.

File: packages/react/src/transactions/useBlockProfiles.ts (hunk 3 above)
Comment (cesarenaldi) [reply packages/react/src/transactions/useBlockProfiles.ts:40]: no, good catch.

**Review by krzysu — APPROVED** [review]

**Pull request merged**